src/state/routing/quickRouteSlice.ts

```diff
@@ -0,0 +1,112 @@
+import { createApi, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react'
+import { sendAnalyticsEvent } from 'analytics'
+import ms from 'ms'
+import { logSwapQuoteRequest } from 'tracing/swapFlowLoggers'
+import { trace } from 'tracing/trace'
+
+import { GetQuickQuoteArgs, PreviewTradeResult, QuickRouteResponse, QuoteState, RouterPreference } from './types'
+import { isExactInput, transformQuickRouteToTrade } from './utils'
+
+const UNISWAP_API_URL = process.env.REACT_APP_UNISWAP_API_URL
+if (UNISWAP_API_URL === undefined) {
+  throw new Error(`UNISWAP_API_URL must be a defined environment variable`)
+}
+
+function getQuoteLatencyMeasure(mark: PerformanceMark): PerformanceMeasure {
+  performance.mark('quickroute-fetch-end')
+  return performance.measure('quickroute-fetch-latency', mark.name, 'quickroute-fetch-end')
+}
+
+export const quickRouteApi = createApi({
+  reducerPath: 'quickRouteApi',
+  baseQuery: fetchBaseQuery({
+    baseUrl: UNISWAP_API_URL,
+  }),
+  endpoints: (build) => ({
+    getQuickRoute: build.query<PreviewTradeResult, GetQuickQuoteArgs>({
+      async onQueryStarted(args: GetQuickQuoteArgs, { queryFulfilled }) {
+        trace(
+          'quickroute',
+          async ({ setTraceError, setTraceStatus }) => {
+            try {
+              await queryFulfilled
+            } catch (error: unknown) {
+              if (error && typeof error === 'object' && 'error' in error) {
+                const queryError = (error as Record<'error', FetchBaseQueryError>).error
+                if (typeof queryError.status === 'number') {
+                  setTraceStatus(queryError.status)
+                }
+                setTraceError(queryError)
+              } else {
+                throw error
+              }
+            }
+          },
+          {
+            data: {
+              ...args,
+            },
+          }
+        )
+      },
+      async queryFn(args, _api, _extraOptions, fetch) {
+        logSwapQuoteRequest(args.tokenInChainId, RouterPreference.API, true)
```

Comment:
Should we always set RouterPreference to API even though as user can have X has their preferred route and still get a quick quote?

Reply:
ah good point